caikit/runtime/http_server/pydantic_wrapper.py

```diff
@@ -170,3 +181,182 @@ def _from_base64(data: Union[bytes, str]) -> bytes:
     if isinstance(data, str):
         return base64.b64decode(data.encode("utf-8"))
     return data
+
+
+async def pydantic_from_request(
+    pydantic_model: Type[pydantic.BaseModel], request: Request
+):
+    """Function to convert a fastapi request into a given pydantic model. This
+    function parses the requests Content-Type and then correctly decodes the data.
+    The currently supported Content-Types are `application/json`
+    and `multipart/form-data`"""
+    content_type = request.headers.get("Content-Type")
+    log.debug4("Detected request using %s type", content_type)
+
+    # If content type is json use pydantic to parse
+    if content_type == "application/json":
+        raw_content = await request.body()
+        try:
+            return pydantic_model.model_validate_json(raw_content)
+        except pydantic.ValidationError as err:
+            raise RequestValidationError(  # pylint: disable=raise-missing-from
```

Comment:
Any reason not to do `from err`?

Reply:
I wasn't sure how fastapi's exception response would handle chained errors. I'll test it and add the chain if nothing breaks